src/library/scala/collection/immutable/HashMap.scala

```diff
@@ -717,19 +722,37 @@ private final class BitmapIndexedMapNode[K, +V](
   }
 
   override def foreach[U](f: ((K, V)) => U): Unit = {
+    val thisPayloadArity = payloadArity
```

Comment:
Having such a long name is not really helpful for code comprehension.  Something like `val iN = payloadArity // arity doesn't change during this operation` is probably clearer.